src/components/CardSection.vue

```diff
@@ -0,0 +1,113 @@
+<script lang="ts">
+import { defineComponent } from 'vue';
+import Card from './Card.vue';
+
+export default defineComponent({
+	name: 'CardSection',
+	components: {
+		Card
+	},
+
+	data() {
+		return {
+			id: 1,
+		}
+	},
+
+	props: {
+		sectionTitle: {
+			type: String,
+			default: '',
+		},
+	},
+
+	computed: {
+		sectionId(): string {
+			if (this.sectionTitle === '') {
+				return `section-${this.id++}`;
+			}
+			return this.sectionTitle.toLowerCase().replace(/\s/g, '-');
+		}
+	}
+});
+</script>
+
+<template>
+	<section :id=sectionId>
+		<h2 v-if="sectionTitle !== ''" class="section-header">{{ sectionTitle }}</h2>
+		<h2 v-else class="screenreader-only">Links</h2>
+		<ul
+			:class="sectionTitle !== '' ? 'card-container' : 'card-container--without-heading'">
+			<slot />
+		</ul>
+	</section>
+</template>
+
+<style scoped lang="scss">
+@import '../styles/breakpoints';
+@import '../styles/utils';
+
+section {
+	border-top: 1px solid var(--colors-accent-deep-secondary);
+	margin: 4rem 0 2rem 0;
+
+	&:first-of-type {
+		border-top: none;
+
+		@media (min-width: $breakpoints-medium) {
+			border-top: 1px solid var(--colors-accent-deep-secondary);
+		}
+	}
+
+	h2 {
+		text-transform: uppercase;
+		font-size: 1.1rem;
+		color: var(--colors-brand-base-secondary);
+		font-family: system-ui, sans-serif;
+		margin-top: 1rem;
+		margin-bottom: 2rem;
+	}
+}
+
+ul {
+	list-style: none;
+	padding: 0;
+	margin: 0;
+	display: grid;
+	gap: 1rem;
+
+	@media (min-width: $breakpoints-medium) {
+		grid-template-columns: 1fr 1fr;
+	}
+
+	@media (min-width: $breakpoints-large) {
+		grid-template-columns: 1fr 1fr 1fr;
+	}
+}
+
+.card-container {
+	&--without-heading {
+		margin-top: 4rem;
+	}
+}
+</style>
+
+<style is:global lang="scss">
```

Comment:
If it's global, why is it defined in a component?

Reply:
Astro let's us use `is:global` in a component to override global styles _when the component is present on the page_. I can't pretend to understand it better than that lol..  🙃

This CSS could have technically just gone into `global.scss` but I wanted to tryout the feature lol

Reply:
Ah I see. I would just add a new css class and scope that in. It will always take your changes. And if those fail, it'll default back to the global ones. Modifying global styles in components seem... hacky to me, but that's a cool feature!